bob-backend/src/pearl/group.rs

```diff
@@ -569,39 +598,72 @@ impl Group {
         });
     }
 
-    pub async fn delete(&self, key: BobKey) -> Result<u64, Error> {
-        let holders = self.holders.read().await;
+    pub async fn delete(
+        &self,
+        key: BobKey,
+        timestamp_config: StartTimestampConfig,
+        force_delete: bool,
+    ) -> Result<u64, Error> {
+        {
+            let holders = self.holders.read().await;
+            if holders.len() != 0 {
+                let copy = holders.iter().cloned().enumerate().collect();
+                drop(holders);
+                return self.delete_in_holders(copy, key, force_delete).await;
+            }
+        }
+        let holder = self
+            .get_actual_holder(get_current_timestamp(), timestamp_config)
+            .await?;
+        self.delete_in_holders(vec![(holder.0, holder.1)], key, force_delete)
```

Comment:
This is unsafe. You should acquire lock on holders and call `delete_in_holders` on full collection. There is no guarantee that the collection will have single item due to the concurrency